### PR TITLE
fix: Make program field read-only in event payload [DHIS2-11225]

### PR DESCRIPTION
**Pull request by enricocolasante**
Program field from event payload is now read-only, it means that is only used for the exporter.
- Completely ignore program field in the importer (Do not load it into the preheat, always getting it from program stage)
- Remove obsolete EventProgramPreProcessor
- Remove obsolete error E1089
- Create the new relevant error E1079
- Update all relevant tests

### CODE REVIEW

File: dhis-2/dhis-e2e-test/src/test/java/org/hisp/dhis/tracker/importer/events/EventValidationTests.java [hunk 1]
```diff
@@ -84,8 +88,7 @@
                 "E1123" ),
             Arguments.arguments( ouIdWithoutAccess, eventProgramId, eventProgramStageId,
                 "E1029" ),
-            Arguments.arguments( OU_ID, trackerProgramId, null, "E1123" ),
-            Arguments.arguments( OU_ID, trackerProgramId, eventProgramStageId, "E1089" ) );
```
Comment (vilkg) [inline dhis-2/dhis-e2e-test/src/test/java/org/hisp/dhis/tracker/importer/events/EventValidationTests.java:88]: Why did you remove this test case? Program stage should still match the program

**Comment by vilkg** [comment]
Will this be backported to 2.36? From the changes in e2e tests it seems that the contract changed a bit, so what will happen after migration to 2.37 for clients who started consuming NTI in 2.36?

File: dhis-2/dhis-e2e-test/src/test/java/org/hisp/dhis/tracker/importer/events/EventValidationTests.java (hunk 1 above)
Comment (enricocolasante) [reply dhis-2/dhis-e2e-test/src/test/java/org/hisp/dhis/tracker/importer/events/EventValidationTests.java:88]: The test case was removed because the program is now ignored in the payload, so the error cannot be triggered anymore.

The other case that now is there is a little be different and it is validating that the program linked by the program stage is the same as the program linked by the enrollment

**Comment by enricocolasante** [comment]
> Will this be backported to 2.36? From the changes in e2e tests it seems that the contract changed a bit, so what will happen after migration to 2.37 for clients who started consuming NTI in 2.36?

I think we need to discuss it with @Bekkalizer  and @stian-sandvold.
This PR has 2 main components in it

- adding missing validation to check that program in program stage and enrollment is the same
- removing program from event payload as it is not really used to persist anything

We should definitely backport the missing validation. The other one in my opinion is an improvement hence it can be just part of 2.37

**Review by gnespolino — APPROVED** [review]

**Review by stian-sandvold — APPROVED** [review]

**Review by ameenhere — APPROVED** [review]
